independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/ArchivePathTree.java

```diff
@@ -224,15 +224,29 @@ public boolean equals(Object obj) {
                 && manifestEnabled == other.manifestEnabled;
     }
 
-    protected class OpenArchivePathTree extends DirectoryPathTree {
+    protected class OpenArchivePathTree extends OpenContainerPathTree {
 
         // we don't make the field final as we want to nullify it on close
-        private volatile FileSystem fs;
+        private volatile boolean open = true;
+        private FileSystem fs;
+        private Path archivePath;
+        private Path rootPath;
         private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
 
-        protected OpenArchivePathTree(FileSystem fs) {
-            super(fs.getPath("/"), pathFilter, ArchivePathTree.this);
+        protected OpenArchivePathTree(Path archive, FileSystem fs) {
```

Comment:
It doesn't look like `archive` needs to be passed in here.

Reply:
Actually it needs to be passed but you're right, I wasn't setting things properly.

Reply:
Why does it need to be passed in?

Reply:
Because of the hashCode/equals. We need something stable that is not leaking for them. The path of the archive makes perfect sense for that.

Reply:
I mean `archive` from the outer class would be enough for that

Reply:
I need it for `SharedOpenArchivePathTree` which extends this class.